apps/frontend/src/app/components/auth/Login.tsx

```diff
@@ -25,77 +25,52 @@ import {
   SignInWithGoogleButton,
 } from './styles';
 import { trpc } from '../../../utils/trpc';
-import { useCallback, useContext, useEffect, useState } from 'react';
-import { validateEmail, validatePassword } from '@dnd-assistant/shared-utils';
+import { useContext, useState } from 'react';
 import { getIonInputClassNames } from './input';
 import { SessionContext } from '../../context/session/SessionContext';
 import { Routes } from '../../routes';
+import { handleTRPCErrors } from '../../../utils/handleTRPCErrors';
 
 export const Login: React.FC = () => {
-  const login = trpc.user.login.useMutation();
+  const [presentToast] = useIonToast();
+  const loginMutation = trpc.user.login.useMutation({
+    onError: (error) => {
+      handleTRPCErrors(error.data?.httpStatus, presentToast, {
+        400: 'The email or password you submited is incorrect.',
+      });
+    },
+  });
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
-  const [toastMessage, setToastMessage] = useState('');
   const [emailIsTouched, setEmailIsTouched] = useState(false);
   const [passwordIsTouched, setPasswordIsTouched] = useState(false);
-  const [emailIsValid, setEmailIsValid] = useState(true);
-  const [passwordIsValid, setPasswordIsValid] = useState(true);
-  const [loginIsDisabled, setLoginIsDisabled] = useState(false);
-  const [showToast, setShowToast] = useState(false);
 
   const { setSession } = useContext(SessionContext);
   const router = useIonRouter();
 
-  const loginHook = useCallback(() => {
-    setLoginIsDisabled(true);
-    login.mutate({
+  const loginHook = () => {
```

Comment:
Since this isn't a hook, but rather attached to a form event, mind changing this to something like `submitLogin`?